src/context/Modal.js

```diff
@@ -0,0 +1,42 @@
+/* eslint-disable react/jsx-props-no-spreading */
+import loadable from '@loadable/component';
+import { useContext } from 'react';
+
+import { ModalDispatchContext, ModalStateContext } from 'context/ModalContext';
+
+// 전체 모달 리스트
+const modalList = {
+  AddForderModal: loadable(() => import('components/Modal/AddFolderModal')),
+  AddToForderModal: loadable(() => import('components/Modal/AddToFolderModal')),
+  DeleteFolderModal: loadable(() => import('components/Modal/DeleteFolderModal')),
+  DeleteLinkModal: loadable(() => import('components/Modal/DeleteLinkModal')),
+  EditFolderNameModal: loadable(() => import('components/Modal/EditFolderNameModal')),
+  ShareModal: loadable(() => import('components/Modal/ShareModal')),
+};
+
+// 다른 곳에서 불러올 모달 컴포넌트
+function Modal() {
+  const openedModal = useContext(ModalStateContext);
+  const { close } = useContext(ModalDispatchContext);
+
+  if (!openedModal) {
+    return null;
+  }
+
+  const { ModalComponent, propList } = openedModal;
+  const { onSubmit, ...restPropList } = propList;
+
+  const onClose = () => {
+    close(ModalComponent);
+  };
+
+  // onSubmit -> 동기/비동기 여부를 모르기 때문에 비동기로 처리
+  const handleSubmit = async () => {
+    await onSubmit();
+    onClose();
+  };
+
+  return <ModalComponent {...restPropList} onSubmit={handleSubmit} onClose={onClose} />;
+}
+
+export { modalList, Modal };
```

Comment:
```suggestion
function Modal() {
  const openedModal = useContext(ModalStateContext);
  const { close } = useContext(ModalDispatchContext);

  if (!openedModal) {
    return null;
  }

  const { ModalComponent, propList } = openedModal;
  const props = propList;

  const onClose = () => {
    close(ModalComponent);
  };

  // onSubmit -> 동기/비동기 여부를 모르기 때문에 비동기로 처리
  const handleSubmit = async () => {
    if (props && props.onSubmit) {
      await props.onSubmit();
    }
    if (onClose) {
      onClose();
    }
  };

  return <ModalComponent {...props} onSubmit={handleSubmit} onClose={onClose} />;
}

export { modalList, Modal };
```
너무 잘 만들어주셨는데, 지금 인자로 주입받은 propsList에서 onSubmit을 구조분해할당하고 있는 부분에서 `onSubmit` 함수가 없는 경우에 대한 예외처리가 필요할거 같아요!

TypeScript를 쓰게 된다면 onSubmit을 필수값으로 지정하는 방식 등을 통해서(물론 여기서도 예외 처리가 필요합니다만) 어느정도 해결은 가능하지만 지금은 JavaScript 환경이고 아시다시피 JavaScript를 인자 주입과 관련해서 제약이 없기 때문에 꼼꼼한 예외처리가 필요합니다!